tests/test_operator.py

```diff
@@ -950,6 +953,37 @@ def test_incomplete_override(self):
 
 class TestDeclarator(object):
 
+    def test_conditional_declarations(self):
+        accesses = [[0, 0], [0, 1], [1, 0], [1, 1]]
+        dims = len(accesses[0])
+        pts = len(accesses)
+        stencil_name = 's%dd_%s_%dpt' % (dims, 'name', pts)
+        stencil_array = Array(
+            name=stencil_name,
+            dimensions=(DefaultDimension(name='len', default_value=dims * pts),),
+            dtype=np.int32,
+            scope='stack'
+        )
+        list_initialize = Expression(ClusterizedEq(Eq(
+            stencil_array,
+            ListInitializer(list(itertools.chain(*accesses)))
+        )))
+
+        iet = Conditional(x < 3, list_initialize, list_initialize)
+
+        parameters = derive_parameters(iet, True)
+        iet = iet_insert_decls(iet, parameters)
+        iet = iet_insert_casts(iet, parameters)
```

Comment:
do we really need this?

Reply:
Insert decls is needed. I'll remove insert casts.